### PR TITLE
ENT-5652: Allow schema property in node.conf

### PR DESCRIPTION
**Pull request by davidjgolding**
The `schema` property for `node.conf` has been introduced to OS, furthering the harmonisation between OS and Enterprise.

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt [hunk 1]
```diff
@@ -33,10 +33,12 @@ const val NODE_DATABASE_PREFIX = "node_"
 
 // This class forms part of the node config and so any changes to it must be handled with care
 data class DatabaseConfig(
+        val schema: String = Defaults.schema,
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt]: I think it is nullable in Enterprise, and defaults to `null` there.  Might as well make the same unless I'm missing something?

Comment (davidjgolding) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt]: Yes that is the case with Enterprise, however, didn't @mnesbit suggest in Monday's standup to default to "sa"?

**Comment by fowlerrr** [comment]
@blsemo can you have a look over this one?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt (hunk 1 above)
Comment (blsemo) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt]: I am not convinced by this - we want to support Postgres as a first class DB as well, and sa is not good schema name for that.

File: tools/demobench/src/main/kotlin/net/corda/demobench/model/NodeConfig.kt [hunk 2]
```diff
@@ -34,8 +34,7 @@ data class NodeConfig(
         /** This is an extra config used by the Cash app. */
         val issuableCurrencies: List<String> = emptyList(),
         /** Pass-through for generating node.conf with external DB */
-        val dataSourceProperties: Properties? = null,
-        val database: Properties? = null,
+        val database: DatabaseConfig = DatabaseConfig(),
```
Comment (blsemo) [inline tools/demobench/src/main/kotlin/net/corda/demobench/model/NodeConfig.kt]: Hmm, how does this deal with `dataSourceProperties` now? Also, why do we now need to know about the `DatabaseConfig` class?

File: tools/demobench/src/test/kotlin/net/corda/demobench/model/NodeConfigTest.kt [hunk 3]
```diff
@@ -130,12 +132,29 @@ class NodeConfigTest {
...
 
+    @Test(timeout=300_000)
+    fun `ensure schema property doesn't throw exception`() {
+        assertDoesNotThrow {
+            createConfig(
+                legalName = myLegalName,
+                p2pPort = 10001,
+                rpcPort = 10002,
+                rpcAdminPort = 10003,
+                database = DatabaseConfig(schema = "sa"),
```
Comment (blsemo) [inline tools/demobench/src/test/kotlin/net/corda/demobench/model/NodeConfigTest.kt]: Does setting schema to "sa" make any difference when the default is already "sa"?

**Review by blsemo — CHANGES_REQUESTED** [review]
I think this would work with H2, not so sure about Postgres. I also have a few question around the changes to demobench.

**Comment by davidjgolding** [comment]
@blsemo I've simplified it and used `null` instead of `sa`

**Comment by blsemo** [comment]
So what happens with the `schema` value now once it has been set? Is it used anywhere?

**Comment by davidjgolding** [comment]
No it's not used anywhere - we assumed it was a missed bit of harmonisation so moved the property over to OS

**Pull request closed without merging**